src/components/AnalysisModal.js

```diff
@@ -62,18 +63,18 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
      */
     const enterNextViewMode = (currentTool, baseViewMode = viewMode) => {
       const app = currentApp(currentTool)
-      const doesCloudEnvForToolExist = currentRuntimeTool === currentTool || app
+      const doesCloudEnvForToolExist = !!(currentRuntimeTool === currentTool || app)
 
       Utils.switchCase(baseViewMode,
         [analysisMode, () => Utils.cond(
           [doesCloudEnvForToolExist, onSuccess],
-          [!doesCloudEnvForToolExist && currentRuntime && isResourceDeletable('runtime', currentRuntime), () => setViewMode(environmentMode)],
+          [!doesCloudEnvForToolExist && !!currentRuntime && isResourceDeletable('runtime', currentRuntime), () => setViewMode(environmentMode)],
           [!doesCloudEnvForToolExist && !currentRuntime, () => setViewMode(environmentMode)],
-          [!doesCloudEnvForToolExist && currentRuntime && !isResourceDeletable('runtime', currentRuntime), onSuccess]
+          [!doesCloudEnvForToolExist && !!currentRuntime && !isResourceDeletable('runtime', currentRuntime), onSuccess]
         )],
         [environmentMode, onSuccess],
         [Utils.DEFAULT, () => Utils.cond(
-          [currentTool === tools.RStudio.label || currentTool === tools.Jupyter.label, () => setViewMode(analysisMode)],
+          [currentTool === tools.RStudio.label || currentTool === tools.Jupyter.label || currentTool === tools.Azure.label, () => setViewMode(analysisMode)],
```

Comment:
these are IDs based on what leonardo labels the runtime's tool, which happen to be the same as the string we display to the user to represent the tool.

Reply:
It looks like `label` is used for both ID and user facing text, which may cause problems because the user facing text can't be reworded without affecting logic.